### ISSUE TITLE
[Request] override template loading mechanism(s)

### ISSUE THREAD

**Issue by r3wt** [twigjs/twig.js#294]
allow replacement of xhr/template loading function.

This would be beneficial to my project, and maybe others as well.

**Comment by plepe** [comment]
To pass the template as "data" parameter is not an option? Or do you need inline includes?

**Comment by r3wt** [comment]
I'm already passing templates as data parameters, but it would be nice to have the xhr implementation settable so that i can take advantage of inline includes as well.

**Comment by plepe** [comment]
I'd suggest you start hacking on it yourself. You can find the call to XMLHttpRequest in `src/twig.core.js`, line 936ff. I'm looking forward to a pull request :-)

**Comment by r3wt** [comment]
10-4 plepe, i'll give it a go!

**Comment by r3wt** [comment]
@plepe  where is Twig's constructor? i'm really having trouble understanding the pattern of this library.

**Comment by plepe** [comment]
I haven't looked into Twig's internals yet either - better ask @justjohn , the founder. But the Twig constructor is in src/twig.core.js, right on top (there's not too much there, though).

**Comment by r3wt** [comment]
@plepe How long do you estimate until this can land? would it be possible to have in next release?

**Comment by r3wt** [comment]
why is this closed? the problem was not actually solved, since its not possible to load template via `{% include %}` in browser! ajax cannot be synchronous.... this really gets me worked up because its such an obviously necessary feature to even make this usuable in the browser. as it is, we are using it in one of our mobile apps with phonegap but the limitation is that we can't use inline includes, which leads to lots of extra code that belongs in subtemplates

**Comment by plepe** [comment]
Sorry, I thought, that #301 solved your issue?

**Comment by r3wt** [comment]
it allowed me to implement a custom loader sure, but there is still the issue of using inline templates. since twig.js apparently can't support asynchronous template loading without losing state, which ultimately made the solution worthless since i gain nothing. i had already implemented a 3rd party loading mechanism for my templates and then render with twig from string.... ultimately this does not provide a way to use inline templates in the browser, which is my only single gripe with twig.js everything else is great and performance is good enough to barely even notice an impact even with large complex templates.

**Comment by plepe** [comment]
Hmm, I see. I created a pull request for tests concerning async inline include loading: #326 .

I'm not sure about internal working of Twig concerning loading inline includes: Maybe they are loaded during render? Than it's clear that loading won't work, because render() is done synchronously. I already looked into rewriting render() asynchronously, but this is a huge effort. And it would break the API, as render() can't return the rendered template, but would require a callback function.
